collector/src/bin/collector.rs

```diff
@@ -196,7 +196,15 @@ async fn collect_data<'a>(
 
     // Setting up Client
     let grpc_timeout = conf.grpc_timeout;
-    let channel = Channel::from_shared(grpc_host)?.timeout(Duration::from_secs(grpc_timeout));
+
+    // Set up a tls client if required using the locally installed trust store.
+    let endpoint = Channel::from_shared(grpc_host)?.timeout(Duration::from_secs(grpc_timeout));
+    let channel = if endpoint.uri().scheme().map_or(false, |x| x == &http::uri::Scheme::HTTPS) {
```

Comment:
```suggestion
    let channel = if endpoint.uri().scheme() == Some(&http::uri::Scheme::HTTPS) {
```

Reply:
So this is resolved, but the code is unchanged. What is the meaning of this?

Reply:
Probably clicked the wrong button by accident